modules/improvedigitalBidAdapter.js

```diff
@@ -192,539 +223,449 @@ export const spec = {
+        switch (nativeItem.assetType) {
+          case NATIVE_DATA.ASSET_TYPES.TITLE:
+            asset.title = {len: cur.len || nativeItem.default.len};
+            break;
+          case NATIVE_DATA.ASSET_TYPES.DATA:
+            asset.data = cleanObj({type: nativeItem.type, len: cur.len})
+            break;
+          case NATIVE_DATA.ASSET_TYPES.IMG:
+            const img = {
+              type: nativeItem.type
+            }
+            if (cur.sizes) {
+              [img.w, img.h] = cur.sizes;
+            } else if (cur.aspect_ratios) {
+              img.wmin = cur.aspect_ratios[0].min_width;
+              img.hmin = cur.aspect_ratios[0].min_height;
+            }
+            asset.img = cleanObj(img);
+            break;
+          default:
+            return;
         }
+        request.assets.push(asset);
+      }
     }
-  };
+    return { ver: NATIVE_DATA.VERSION, request: JSON.stringify(request) };
+  },
 
-  this.createBasicBidRequestObject = function(requestParameters, extraRequestParameters) {
-    let impressionBidRequestObject = {};
-    impressionBidRequestObject.secure = 1;
-    if (requestParameters.requestId) {
-      impressionBidRequestObject.id = requestParameters.requestId;
+  makeServerRequest(requestObject, bidRequests, bidderRequest) {
+    const requests = [];
+    if (config.getConfig('improvedigital.singleRequest') === true) {
+      requestObject.imp = bidRequests.map(this.buildImp);
+      requests[0] = this.formatRequest(requestObject, bidderRequest);
     } else {
-      impressionBidRequestObject.id = getUniqueIdentifierStr();
-    }
-    if (requestParameters.domain) {
-      impressionBidRequestObject.domain = requestParameters.domain;
-    }
-    if (requestParameters.page) {
-      impressionBidRequestObject.page = requestParameters.page;
-    }
-    if (requestParameters.ref) {
-      impressionBidRequestObject.ref = requestParameters.ref;
+      bidRequests.map((bidRequest) => {
+        const request = deepClone(requestObject);
+        request.id = bidRequest.bidId || getUniqueIdentifierStr();
+        request.imp = [ID_REQUEST.buildImp(bidRequest)];
+        if (bidRequest.transactionId) {
+          deepSetValue(request, 'source.tid', bidRequest.transactionId);
+        }
+        requests.push(this.formatRequest(request, bidderRequest));
+      });
     }
-    if (requestParameters.callback) {
-      impressionBidRequestObject.callback = requestParameters.callback;
+
+    return requests;
+  },
+
+  formatRequest(request, bidderRequest) {
+    return {
+      method: 'POST',
+      url: REQUEST_URL,
+      data: JSON.stringify(request),
+      bidderRequest
     }
-    if (requestParameters.libVersion) {
-      impressionBidRequestObject.version = requestParameters.libVersion + '-' + this.CONSTANTS.CLIENT_VERSION;
+  },
+};
+
+const ID_RESPONSE = {
+  buildAd(bidObject, bid, bidRequest, bidResponse) {
+    if (deepAccess(bidObject, 'mediaTypes.video')) {
```

Comment:
this logic only works when `mediaTypes` contains only one type. then the type of creative in the response will match the mediatType of the ad unit. However, what about multi-format ad units? There the only option right now is to check the adm field.

Please change the logic as follows:
* if only one mediatype, then the current logic
* else check the adm field like this:
if adm .search(/^<vast/i) === 0 -> buildVideoAd
else if adm indexOf('{'}) === 0 -> buildNativeAd
else buildBannerAd